### ISSUE TITLE
[Bug]: Newest commit throws error when hitting Generate

### ISSUE THREAD

**Issue by steyRNG** [lshqqytiger/stable-diffusion-webui-directml#77]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What happened?

Can't generate Pictures anymore with newest commit, throws error instantly when hitting "Generate" and doesn't output pictures. When going back to commit dae3394497ea810a4601422e8c7f3c74debe4c7c everything works as usual.

### Steps to reproduce the problem

Update to newest commit,
start webui, hit generate, observe error in console.

### What should have happened?

Pictures Generated like normal

### Commit where the problem happens

b48ae2ba41a293bee9aaa22afe7a2b7e39a39fe0

### What platforms do you use to access the UI ?

Windows

### What browsers do you use to access the UI ?

Google Chrome

### Command Line Arguments

```Shell
--opt-sub-quad-attention --no-half --disable-nan-check --opt-split-attention-v1 --precision full
```

### List of extensions

<img width="515" alt="image" src="https://user-images.githubusercontent.com/123616306/232278595-40d0e702-1e6d-47d0-b370-7330051902c2.png">

### Console logs

```Shell
Error completing request
Arguments: ('task(0gbg9qoxgznzybv)', '', '', [], 28, 0, False, False, 1, 1, 7, -1.0, -1.0, 0, 0, 0, False, 512, 512, False, 0.7, 2, 'Latent', 0, 0, 0, [], 0, <scripts.external_code.ControlNetUnit object at 0x0000014FC8AB6FB0>, <scripts.external_code.ControlNetUnit object at 0x0000014FC8AB6FE0>, False, '', 0.5, True, False, '', 'Lerp', False, False, False, 'Horizontal', '1,1', '0.2', False, False, False, 'Attention', False, False, False, False, False, 'positive', 'comma', 0, False, False, '', 1, '', 0, '', 0, '', True, False, False, False, 0, None, False, None, False, 50) {}
Traceback (most recent call last):
  File "G:\Other\AI\stable-diffusion-webui-directml\modules\call_queue.py", line 56, in f
    res = list(func(*args, **kwargs))
  File "G:\Other\AI\stable-diffusion-webui-directml\modules\call_queue.py", line 37, in f
    res = func(*args, **kwargs)
  File "G:\Other\AI\stable-diffusion-webui-directml\modules\txt2img.py", line 56, in txt2img
    processed = process_images(p)
  File "G:\Other\AI\stable-diffusion-webui-directml\modules\processing.py", line 503, in process_images
    res = process_images_inner(p)
  File "G:\Other\AI\stable-diffusion-webui-directml\modules\processing.py", line 653, in process_images_inner
    samples_ddim = p.sample(conditioning=c, unconditional_conditioning=uc, seeds=seeds, subseeds=subseeds, subseed_strength=p.subseed_strength, prompts=prompts)
  File "G:\Other\AI\stable-diffusion-webui-directml\modules\processing.py", line 869, in sample
    samples = self.sampler.sample(self, x, conditioning, unconditional_conditioning, image_conditioning=self.txt2img_image_conditioning(x))
  File "G:\Other\AI\stable-diffusion-webui-directml\modules\sd_samplers_kdiffusion.py", line 358, in sample
    samples = self.launch_sampling(steps, lambda: self.func(self.model_wrap_cfg, x, extra_args={
  File "G:\Other\AI\stable-diffusion-webui-directml\modules\sd_samplers_kdiffusion.py", line 234, in launch_sampling
    return func()
  File "G:\Other\AI\stable-diffusion-webui-directml\modules\sd_samplers_kdiffusion.py", line 358, in <lambda>
    samples = self.launch_sampling(steps, lambda: self.func(self.model_wrap_cfg, x, extra_args={
  File "G:\Other\AI\stable-diffusion-webui-directml\venv\lib\site-packages\torch\autograd\grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "G:\Other\AI\stable-diffusion-webui-directml\repositories\k-diffusion\k_diffusion\sampling.py", line 150, in sample_euler_ancestral
    denoised = model(x, sigmas[i] * s_in, **extra_args)
  File "G:\Other\AI\stable-diffusion-webui-directml\venv\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "G:\Other\AI\stable-diffusion-webui-directml\modules\sd_samplers_kdiffusion.py", line 126, in forward
    x_out = self.inner_model(x_in, sigma_in, cond=make_condition_dict([cond_in], image_cond_in))
  File "G:\Other\AI\stable-diffusion-webui-directml\venv\lib\site-packages\torch\nn\modules\module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "G:\Other\AI\stable-diffusion-webui-directml\repositories\k-diffusion\k_diffusion\external.py", line 112, in forward
    eps = self.get_eps(input * c_in, self.sigma_to_t(sigma), **kwargs)
  File "G:\Other\AI\stable-diffusion-webui-directml\repositories\k-diffusion\k_diffusion\external.py", line 72, in sigma_to_t
    low_idx = dists.ge(0).cumsum(dim=0).argmax(dim=0).clamp(max=self.log_sigmas.shape[0] - 2)
RuntimeError: Falscher Parameter.
```

### Additional information

CPU: 5950x
RAM: 32GB
GPU: 7900 XTX
VRAM: 24GB

**Comment by odraMorningstar** [comment]
its the recent commit, you can change the version with this commit git checkout dae3394497ea810a4601422e8c7f3c74debe4c7c

**Comment by hanzoro5** [comment]
The last two updates break everything. The webUI interface does not function properly, now I can't even generate an image anymore.

**Comment by Lumpsack** [comment]
Go back to the old version of devices.py, it should work then (or at least it did for me) - by the way this issue is a duplicate of the one above it in the list currently.

**Comment by lshqqytiger** [comment]
Please run `pip show torch-directml` and copy output here.

**Comment by steyRNG** [comment]
> Please run `pip show torch-directml` and copy output here.

PS G:\Other\AI\stable-diffusion-webui-directml> pip show torch-directml
WARNING: Package(s) not found: torch-directml
PS G:\Other\AI\stable-diffusion-webui-directml> .\venv\Scripts\activate
(venv) PS G:\Other\AI\stable-diffusion-webui-directml> pip show torch-directml
Name: torch-directml
Version: 0.1.13.1.dev230301
Summary: A DirectML backend for hardware acceleration in PyTorch.
Home-page: https://github.com/microsoft/directml
Author: Microsoft Corporation
Author-email: askdirectml@microsoft.com
License: MIT
Location: g:\other\ai\stable-diffusion-webui-directml\venv\lib\site-packages
Requires: torch, torchvision
Required-by:

**Comment by lshqqytiger** [comment]
Run `pip install torch-directml --upgrade` and try again.
Latest commit does not support older version of DirectML anymore.

**Comment by steyRNG** [comment]
Okay, works with updated Torch-DirectML. Was there any announcement i missed? Since it doesn't get updated automatically and it isn't in the requirements.txt i missed updating it as well before reporting

**Comment by lshqqytiger** [comment]
I forgot to mention this change in commit because I automatically checks for update of `torch-directml`.
I'll add a warning in commit when next update releases.

**Comment by Lumpsack** [comment]
Hi,

I tried the upgrade and check then copied back the device.py file - same error for me:, I've included the current results - is there something else that needs changing?

Cheers

-----------------------------------------------------------------------------------------

D:\GitHub\stable-diffusion-webui-directml\modules>pip show torch-directml
Name: torch-directml
Version: 0.1.13.1.dev230413
Summary: A DirectML backend for hardware acceleration in PyTorch.
Home-page: https://github.com/microsoft/directml
Author: Microsoft Corporation
Author-email: askdirectml@microsoft.com
License: MIT
Location: c:\users\<username>\appdata\local\programs\python\python310\lib\site-packages
Requires: torch, torchvision
Required-by:

D:\GitHub\stable-diffusion-webui-directml\modules>pip install torch-directml --upgrade
Requirement already satisfied: torch-directml in c:\users\<username>\appdata\local\programs\python\python310\lib\site-packages (0.1.13.1.dev230413)
Requirement already satisfied: torch==1.13.1 in c:\users\<username>\appdata\local\programs\python\python310\lib\site-packages (from torch-directml) (1.13.1)
Requirement already satisfied: torchvision==0.14.1 in c:\users\<username>\appdata\local\programs\python\python310\lib\site-packages (from torch-directml) (0.14.1)
Requirement already satisfied: typing-extensions in c:\users\<username>\appdata\local\programs\python\python310\lib\site-packages (from torch==1.13.1->torch-directml) (4.5.0)
Requirement already satisfied: numpy in c:\users\<username>\appdata\local\programs\python\python310\lib\site-packages (from torchvision==0.14.1->torch-directml) (1.24.2)
Requirement already satisfied: requests in c:\users\<username>\appdata\local\programs\python\python310\lib\site-packages (from torchvision==0.14.1->torch-directml) (2.28.2)
Requirement already satisfied: pillow!=8.3.*,>=5.3.0 in c:\users\<username>\appdata\local\programs\python\python310\lib\site-packages (from torchvision==0.14.1->torch-directml) (9.5.0)
Requirement already satisfied: charset-normalizer<4,>=2 in c:\users\<username>\appdata\local\programs\python\python310\lib\site-packages (from requests->torchvision==0.14.1->torch-directml) (3.1.0)
Requirement already satisfied: idna<4,>=2.5 in c:\users\<username>\appdata\local\programs\python\python310\lib\site-packages (from requests->torchvision==0.14.1->torch-directml) (3.4)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in c:\users\<username>\appdata\local\programs\python\python310\lib\site-packages (from requests->torchvision==0.14.1->torch-directml) (1.26.15)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\<username>\appdata\local\programs\python\python310\lib\site-packages (from requests->torchvision==0.14.1->torch-directml) (2022.12.7)

**Comment by Lumpsack** [comment]
For me, I trashed the old install and installed from fresh - working now, if a bit painful

**Comment by TeutonJon78** [comment]
I had a similar problem, and didn't want to convert everything over to a fresh install. So i tried just deleting the venv folder. When I started it again, it re-created it and everything starting generating just fine.

That's likely a round about way of running the updated command the is listed.

**Comment by mr-september** [comment]
@lshqqytiger Where do you run `pip install torch-directml --upgrade`? Both windows `cmd` and VS Code apparently don't have `torch-directml` at all, so I'm probably not in the right place/folder/environment?

@TeutonJon78 I tried that, but on the first run, almost every package elected to "Using cached <package-version>" instead of downloading/updating.

**Comment by FreeBlues** [comment]
> 

Thanks, it works after renaming the venv folder!

**Comment by megazaz80** [comment]
> The last two updates break everything. The webUI interface does not function properly, now I can't even generate an image anymore.

For me was broken too, then I did something really strange I copy paste my older SD backup over the new one, then something strange happen )) my speeds go over the roof, before for generating 512x720 with 30 steps one step was over 6 seconds now I getting 1.18 secs per step btw my card is rx6600xt

**Comment by megazaz80** [comment]
![IMG_20230521_105215879_HDR~2](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/70824762/9fdde4e8-ad8e-441c-bce2-b56469cd793b)
